public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -53,48 +49,42 @@ import { RuleEditorSection } from './RuleEditorSection';
 export const MIN_TIME_RANGE_STEP_S = 10; // 10 seconds
 export const MAX_GROUP_RESULTS = 1000;
 
-export const useFolderGroupOptions = (folderUid: string, enableProvisionedGroups: boolean) => {
+const useFetchGroupsForFolder = (folderUid: string) => {
   // fetch the ruler rules from the database so we can figure out what other "groups" are already defined
   // for our folders
-  const { isLoading: isLoadingRulerNamespace, currentData: rulerNamespace } =
-    alertRuleApi.endpoints.rulerNamespace.useQuery(
-      {
-        namespace: folderUid,
-        rulerConfig: GRAFANA_RULER_CONFIG,
-      },
-      {
-        skip: !folderUid,
-        refetchOnMountOrArgChange: true,
-      }
-    );
-
-  // There should be only one entry in the rulerNamespace object
-  // However it uses folder name as key, so to avoid fetching folder name, we use Object.values
-  const groupOptions = useMemo(() => {
-    if (!rulerNamespace) {
-      // still waiting for namespace information to be fetched
-      return [];
+  return alertRuleApi.endpoints.rulerNamespace.useQuery(
+    {
+      namespace: folderUid,
+      rulerConfig: GRAFANA_RULER_CONFIG,
+    },
+    {
+      refetchOnMountOrArgChange: true,
     }
+  );
+};
 
-    const folderGroups = Object.values(rulerNamespace).flat() ?? [];
-
-    return folderGroups
-      .map<SelectableValue<string>>((group) => {
-        const isProvisioned = isProvisionedGroup(group);
-        return {
-          label: group.name,
-          value: group.name,
-          description: group.interval ?? DEFAULT_GROUP_EVALUATION_INTERVAL,
-          // we include provisioned folders, but disable the option to select them
-          isDisabled: !enableProvisionedGroups ? isProvisioned : false,
-          isProvisioned: isProvisioned,
-        };
-      })
-
-      .sort(sortByLabel);
-  }, [rulerNamespace, enableProvisionedGroups]);
-
-  return { groupOptions, loading: isLoadingRulerNamespace };
+const namespaceToGroupOptions = (rulerNamespace: RulerRulesConfigDTO, enableProvisionedGroups: boolean) => {
+  if (!rulerNamespace) {
+    // still waiting for namespace information to be fetched
+    return [];
+  }
+
+  const folderGroups = Object.values(rulerNamespace).flat() ?? [];
```

Comment:
(based on the types) The `if` would always be falsy, as rulerNamespace is required. And then also the `??` operator isn't required as the first part will always be truthy